accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,375 @@
+  
+  #### Control plane and worker nodes
+
+  #### Registering control plane and worker to SUSE Manager
+
+  The core of the CaaS Platform is control plane and worker nodes. The Management Node, instructed by SUSE Manager, will take care of bootstrapping control plane and worker nodes.
+
+  The critical part of every control plane and worker node is handling updates of the underlying Kubernetes-related packages (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`): these updates must be handled with `skuba` from the Management Node.
+
+  [`skuba-update` will take care of updating the Base Operating System](https://susedoc.github.io/doc-caasp/beta/caasp-admin/single-html/#_base_os).
+  
+  #### Bootstrapping control plane and worker nodes
+
+  In SUSE Manager under "CaaS Platform > Deployment > cluster name" there will be a list of machines that:
+  - are not part of any other CaaS Platform cluster
+  - are not already bootstrapped into the current cluster
+  - satisfy the CaaS Platform node requirements: those machines are either bootstrapped via the AutoYaST profile or are already registered with SUSE Manager and satisfy the node requirements (checked via grains).
+
+  The user will then have the option to bootstrap the selected machine as the first control plane (or subsequently, as an additional control plane or worker). Upon triggering of this event, the SUSE Manager will:
+
+  - operate on the to-be control plane or worker node:
+    - Copy the SSH key of the Management Node for passwordless login in the node. In case of existing machines, the SSH file must be copied (e.g. via Salt), whereas the AutoYaST case will be already covered during provisioning
+    - Assign the CaaS Platform channels assigned via Salt (SUSE CaaS Platform Extension to SUSE Linux Enterprise 15 and SUSE Containers Module 15 SP1 x86_64)
+    - Register the machine from SUSE Manager (if it is not registered). In this case, the node gains all the advantages of being managed by SUSE Manager (e.g. access to mirrored channels specified above)
+
+NOTE - Channel syncing: SUSE Manager must sync __all__ the packages that constitute the `patterns-caasp-Node` pattern even in the case of package pinning or other user's customizations. Being in a situation where `skuba` is updated while `patterns-caasp-Node` can lead to problems.
+
+NOTE - Minion registration: it must be clear in the documentation and the user interface of SUSE Manager that the user **cannot** upgrade or reboot the node via SUSE Manager: issuing those actions might result in the malfunction of the cluster. As an additional countermeasure, SUSE Manager can take advantage of the [Minion Blackout Configuration](https://docs.saltstack.com/en/latest/topics/blackout/) or the [`zypper` package locking](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.zypperpkg.html#salt.modules.zypperpkg.add_lock) to restrict the package management of the cluster node.
```

Comment:
I think we should make a complete list of these *critical* restrictions (minion cleanup on delete, remote cmds, etc.) and effectively restrict them in the UI and the API. Since these actions are usually just a few clicks away, even though documented, it's very easy to make a mistake and I imagine it could have a serious impact on production systems.

I think the "minion blackout" feature is a good candidate to implement this. It looks flexible enough (through "whitelist" pillar) to define different sets of allowed actions for different node/cluster types.

Reply:
OK, added a note about it.